tests/test_search_dimensions.py

```diff
@@ -61,7 +61,7 @@ def test_list_behavior(self):
         self.assertEqual(count, 3)
 
         with self.assertRaises(IndexError):
-            dim3 = dims[3]
```

Comment:
I understand the change gets rid of the codeQL error, but it still seems to me like the line isn't doing anything

Reply:
The line is reading an out-of-range index, which should raise an IndexError assert. This change looks correct to me.